<!DOCTYPE UI><UI version="3.3" stdsetdef="1">
<class>MyPasswordSafe</class>
<widget class="QMainWindow">
    <property name="name">
        <cstring>MyPasswordSafe</cstring>
    </property>
    <property name="geometry">
        <rect>
            <x>0</x>
            <y>0</y>
            <width>322</width>
            <height>383</height>
        </rect>
    </property>
    <property name="sizePolicy">
        <sizepolicy>
            <hsizetype>5</hsizetype>
            <vsizetype>5</vsizetype>
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
        </sizepolicy>
    </property>
    <property name="baseSize">
        <size>
            <width>240</width>
            <height>320</height>
        </size>
    </property>
    <property name="caption">
        <string>MyPasswordSafe</string>
    </property>
    <property name="icon">
        <pixmap>Padlock.xpm</pixmap>
    </property>
    <hbox>
        <property name="name">
            <cstring>unnamed</cstring>
        </property>
        <property name="margin">
            <number>0</number>
        </property>
        <widget class="SafeListView">
            <property name="name">
                <cstring>pwordListView</cstring>
            </property>
            <property name="sizePolicy">
                <sizepolicy>
                    <hsizetype>7</hsizetype>
                    <vsizetype>7</vsizetype>
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                </sizepolicy>
            </property>
            <property name="whatsThis" stdset="0">
                <string>This is the password list. A few operations can be performed with the mouse and keyboard. These are:
- Double clicking an item or pressing the space bar when one is active copies the password to the clipboard
- Right clicking an item pops up the password menu
- Pressing return copies the selected item to the clipboard.

</string>
            </property>
        </widget>
    </hbox>
</widget>
<menubar>
    <property name="name">
        <cstring>menubar</cstring>
    </property>
    <item text="&amp;File" name="fileMenu">
        <action name="fileNewAction"/>
        <action name="fileOpenAction"/>
        <action name="fileOpenDefaultAction"/>
        <separator/>
        <action name="fileSaveAction"/>
        <action name="fileSaveAsAction"/>
        <separator/>
        <action name="changePassPhraseAction"/>
        <action name="fileMakeDefaultAction"/>
        <action name="lockAction"/>
        <separator/>
        <action name="fileExitAction"/>
    </item>
    <item text="&amp;Edit" name="PopupMenu">
        <action name="editUndoAction"/>
        <action name="editRedoAction"/>
        <separator/>
        <action name="pwordAddAction"/>
        <action name="createGroupAction"/>
        <action name="pwordDeleteAction"/>
        <action name="pwordEditAction"/>
        <separator/>
        <action name="pwordFetchUserAction"/>
        <action name="pwordFetchAction"/>
    </item>
    <item text="&amp;Settings" name="PopupMenu_2">
        <action name="clearClipboardOnExitAction"/>
        <action name="lockOnMinimizeAction"/>
        <separator/>
        <action name="filePrefAction"/>
    </item>
    <item text="&amp;Help" name="helpMenu">
        <action name="helpWhatsThisAction"/>
        <action name="helpManualAction"/>
        <separator/>
        <action name="helpAboutAction"/>
    </item>
</menubar>
<toolbars>
    <toolbar dock="2">
        <property name="name">
            <cstring>toolBar</cstring>
        </property>
        <property name="label">
            <string>Tools</string>
        </property>
        <action name="fileNewAction"/>
        <action name="fileOpenAction"/>
        <action name="fileSaveAction"/>
        <separator/>
        <action name="createGroupAction"/>
        <action name="pwordAddAction"/>
        <action name="pwordDeleteAction"/>
        <action name="pwordEditAction"/>
        <action name="pwordFetchAction"/>
        <action name="pwordFetchUserAction"/>
        <separator/>
        <action name="lockAction"/>
    </toolbar>
</toolbars>
<customwidgets>
    <customwidget>
        <class>SafeListView</class>
        <header location="global">/home/nolan/src/MyPasswordSafe/src/safelistview.hpp</header>
        <sizehint>
            <width>-1</width>
            <height>-1</height>
        </sizehint>
        <container>0</container>
        <sizepolicy>
            <hordata>5</hordata>
            <verdata>5</verdata>
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
        </sizepolicy>
        <pixmap>image0</pixmap>
        <signal>contextMenuRequested(QListViewItem *, const QPoint &amp;, int)</signal>
        <signal>doubleClicked(QListViewItem *)</signal>
    </customwidget>
</customwidgets>
<actions>
    <action>
        <property name="name">
            <cstring>fileNewAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>filenew2.png</iconset>
        </property>
        <property name="text">
            <string>New</string>
        </property>
        <property name="menuText">
            <string>&amp;New</string>
        </property>
        <property name="statusTip">
            <string>Creates a new safe</string>
        </property>
        <property name="whatsThis">
            <string></string>
        </property>
        <property name="accel">
            <string>Ctrl+N</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>fileOpenAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>fileopen.png</iconset>
        </property>
        <property name="text">
            <string>Open</string>
        </property>
        <property name="menuText">
            <string>&amp;Open...</string>
        </property>
        <property name="statusTip">
            <string>Opens a safe</string>
        </property>
        <property name="accel">
            <string>Ctrl+O</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>fileSaveAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>filesave.png</iconset>
        </property>
        <property name="text">
            <string>Save</string>
        </property>
        <property name="menuText">
            <string>&amp;Save</string>
        </property>
        <property name="statusTip">
            <string>Save the safe</string>
        </property>
        <property name="accel">
            <string>Ctrl+S</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>fileSaveAsAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>filesaveas.png</iconset>
        </property>
        <property name="text">
            <string>Save As</string>
        </property>
        <property name="menuText">
            <string>Save &amp;As...</string>
        </property>
        <property name="statusTip">
            <string>Save the safe to a different file</string>
        </property>
        <property name="accel">
            <string>Ctrl+Shift+S</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>fileExitAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>exit.png</iconset>
        </property>
        <property name="text">
            <string>Exit</string>
        </property>
        <property name="menuText">
            <string>E&amp;xit</string>
        </property>
        <property name="whatsThis">
            <string>Causes MyPasswordSafe to quit.</string>
        </property>
        <property name="accel">
            <string>Ctrl+Q</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>helpAboutAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>info.png</iconset>
        </property>
        <property name="text">
            <string>About</string>
        </property>
        <property name="menuText">
            <string>&amp;About</string>
        </property>
        <property name="statusTip">
            <string>About MyPasswordSafe</string>
        </property>
        <property name="accel">
            <string></string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>pwordAddAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>edit_add.png</iconset>
        </property>
        <property name="text">
            <string>Add Password</string>
        </property>
        <property name="menuText">
            <string>&amp;Add Password</string>
        </property>
        <property name="statusTip">
            <string>Creates a new password</string>
        </property>
        <property name="whatsThis">
            <string></string>
        </property>
        <property name="accel">
            <string>Ins</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>pwordDeleteAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>edit_remove.png</iconset>
        </property>
        <property name="text">
            <string>Delete</string>
        </property>
        <property name="menuText">
            <string>&amp;Delete</string>
        </property>
        <property name="statusTip">
            <string>Deletes the selected item</string>
        </property>
        <property name="accel">
            <string>Del</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>pwordEditAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>edit.png</iconset>
        </property>
        <property name="text">
            <string>Edit Password</string>
        </property>
        <property name="menuText">
            <string>&amp;Edit Password</string>
        </property>
        <property name="statusTip">
            <string>Edits the selected item</string>
        </property>
        <property name="accel">
            <string>Return</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>pwordFetchAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>editcut.png</iconset>
        </property>
        <property name="text">
            <string>Fetch Password</string>
        </property>
        <property name="menuText">
            <string>Fetch &amp;Password</string>
        </property>
        <property name="statusTip">
            <string>Copy the selected password to the clipboard</string>
        </property>
        <property name="accel">
            <string>Space</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>fileMakeDefaultAction</cstring>
        </property>
        <property name="toggleAction">
            <bool>false</bool>
        </property>
        <property name="text">
            <string>&amp;Make default</string>
        </property>
        <property name="menuText">
            <string>&amp;Make default</string>
        </property>
        <property name="statusTip">
            <string>Sets the safe that gets opened when MyPasswordSafe starts.</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>filePrefAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>configure.png</iconset>
        </property>
        <property name="text">
            <string>Preferences</string>
        </property>
        <property name="menuText">
            <string>&amp;Preferences</string>
        </property>
        <property name="statusTip">
            <string>Edit your preferences</string>
        </property>
        <property name="accel">
            <string>Ctrl+P</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>helpWhatsThisAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>contexthelp.png</iconset>
        </property>
        <property name="text">
            <string>What's This?</string>
        </property>
        <property name="menuText">
            <string>What's This?</string>
        </property>
        <property name="accel">
            <string>Shift+F1</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>pwordFetchUserAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>fetch_user.png</iconset>
        </property>
        <property name="text">
            <string>Fetch User</string>
        </property>
        <property name="menuText">
            <string>&amp;Fetch User</string>
        </property>
        <property name="statusTip">
            <string>Copies the username to the clipboard.</string>
        </property>
        <property name="accel">
            <string>Ctrl+U</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>helpManualAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>help.png</iconset>
        </property>
        <property name="text">
            <string>Manual</string>
        </property>
        <property name="menuText">
            <string>&amp;Manual</string>
        </property>
        <property name="statusTip">
            <string>Opens the manual</string>
        </property>
        <property name="accel">
            <string>F1</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>fileOpenDefaultAction</cstring>
        </property>
        <property name="text">
            <string>Open Default</string>
        </property>
        <property name="menuText">
            <string>Open &amp;Default</string>
        </property>
        <property name="statusTip">
            <string>Open the default safe</string>
        </property>
        <property name="accel">
            <string>Ctrl+Shift+O</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>changePassPhraseAction</cstring>
        </property>
        <property name="text">
            <string>Change pass-phrase</string>
        </property>
        <property name="menuText">
            <string>Change &amp;pass-phrase</string>
        </property>
        <property name="statusTip">
            <string>Change the safe's pass-phrase</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>editUndoAction</cstring>
        </property>
        <property name="enabled">
            <bool>false</bool>
        </property>
        <property name="iconSet">
            <iconset>undo.png</iconset>
        </property>
        <property name="text">
            <string>Undo</string>
        </property>
        <property name="menuText">
            <string>&amp;Undo</string>
        </property>
        <property name="statusTip">
            <string>Undo the last thing you did</string>
        </property>
        <property name="accel">
            <string>Ctrl+Z</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>editRedoAction</cstring>
        </property>
        <property name="enabled">
            <bool>false</bool>
        </property>
        <property name="iconSet">
            <iconset>redo.png</iconset>
        </property>
        <property name="text">
            <string>Redo</string>
        </property>
        <property name="menuText">
            <string>&amp;Redo</string>
        </property>
        <property name="statusTip">
            <string>Redo what was undone</string>
        </property>
        <property name="accel">
            <string>Ctrl+Y</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>lockAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>encrypted.png</iconset>
        </property>
        <property name="text">
            <string>Lock MyPasswordSafe</string>
        </property>
        <property name="menuText">
            <string>&amp;Lock MyPasswordSafe</string>
        </property>
        <property name="accel">
            <string>Ctrl+L</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>lockOnMinimizeAction</cstring>
        </property>
        <property name="toggleAction">
            <bool>true</bool>
        </property>
        <property name="text">
            <string>Lock on minimize?</string>
        </property>
        <property name="menuText">
            <string>&amp;Lock on minimize?</string>
        </property>
        <property name="statusTip">
            <string>Lock on minimize?</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>createGroupAction</cstring>
        </property>
        <property name="iconSet">
            <iconset>folder_new.png</iconset>
        </property>
        <property name="text">
            <string>Create Group</string>
        </property>
        <property name="menuText">
            <string>Create &amp;Group</string>
        </property>
        <property name="statusTip">
            <string>Create a new group</string>
        </property>
        <property name="accel">
            <string>Ctrl+G</string>
        </property>
    </action>
    <action>
        <property name="name">
            <cstring>clearClipboardOnExitAction</cstring>
        </property>
        <property name="toggleAction">
            <bool>true</bool>
        </property>
        <property name="text">
            <string>Clear clipboard on exit?</string>
        </property>
        <property name="menuText">
            <string>&amp;Clear clipboard on exit?</string>
        </property>
    </action>
</actions>
<images>
    <image name="image0">
        <data format="XBM.GZ" length="79">789c534e494dcbcc4b554829cdcdad8c2fcf4c29c95030e0524611cd48cd4ccf28010a1797249664262b2467241641a592324b8aa363156c15aab914146aadb90067111b1f</data>
    </image>
</images>
<connections>
    <connection>
        <sender>fileNewAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>fileNew()</slot>
    </connection>
    <connection>
        <sender>fileOpenAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>fileOpen()</slot>
    </connection>
    <connection>
        <sender>fileSaveAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>save()</slot>
    </connection>
    <connection>
        <sender>fileSaveAsAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>saveAs()</slot>
    </connection>
    <connection>
        <sender>fileExitAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>fileExit()</slot>
    </connection>
    <connection>
        <sender>helpAboutAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>helpAbout()</slot>
    </connection>
    <connection>
        <sender>pwordAddAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>pwordAdd()</slot>
    </connection>
    <connection>
        <sender>pwordDeleteAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>pwordDelete()</slot>
    </connection>
    <connection>
        <sender>pwordEditAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>pwordEdit()</slot>
    </connection>
    <connection>
        <sender>pwordFetchAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>pwordFetch()</slot>
    </connection>
    <connection>
        <sender>fileMakeDefaultAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>fileMakeDefault()</slot>
    </connection>
    <connection>
        <sender>filePrefAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>filePreferences()</slot>
    </connection>
    <connection>
        <sender>pwordListView</sender>
        <signal>doubleClicked(QListViewItem*)</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>onPwordListDblClicked(QListViewItem*)</slot>
    </connection>
    <connection>
        <sender>pwordListView</sender>
        <signal>contextMenuRequested(QListViewItem*,const QPoint&amp;,int)</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>onPwordListRightClk(QListViewItem*,const QPoint&amp;,int)</slot>
    </connection>
    <connection>
        <sender>helpWhatsThisAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>whatsThis()</slot>
    </connection>
    <connection>
        <sender>pwordFetchUserAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>pwordFetchUser()</slot>
    </connection>
    <connection>
        <sender>helpManualAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>helpManual()</slot>
    </connection>
    <connection>
        <sender>fileOpenDefaultAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>fileOpenDefault()</slot>
    </connection>
    <connection>
        <sender>changePassPhraseAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>fileChangePassPhrase()</slot>
    </connection>
    <connection>
        <sender>lockAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>lock()</slot>
    </connection>
    <connection>
        <sender>createGroupAction</sender>
        <signal>activated()</signal>
        <receiver>MyPasswordSafe</receiver>
        <slot>createGroup()</slot>
    </connection>
</connections>
<includes>
    <include location="local" impldecl="in implementation">aboutdlg.h</include>
    <include location="local" impldecl="in implementation">preferencesdlg.h</include>
    <include location="global" impldecl="in declaration">string</include>
    <include location="local" impldecl="in declaration">safe.hpp</include>
    <include location="global" impldecl="in declaration">qsettings.h</include>
    <include location="local" impldecl="in declaration">manualdlg.h</include>
    <include location="local" impldecl="in implementation">safelistview.hpp</include>
    <include location="local" impldecl="in implementation">mypasswordsafe.ui.h</include>
</includes>
<forwards>
    <forward>class QListViewItem</forward>
</forwards>
<variables>
    <variable access="private">ManualDlg m_manual;</variable>
    <variable access="private">int m_name_width, m_user_width, m_notes_width, m_max_tries;</variable>
    <variable access="private">QString m_default_safe;</variable>
    <variable access="private">bool m_clear_clipboard;</variable>
    <variable access="private">QString m_def_user;</variable>
    <variable access="private">Safe *m_safe;</variable>
    <variable access="private">unsigned int m_gen_pword_length;</variable>
    <variable access="private">QSettings m_config;</variable>
    <variable access="private">bool m_shown;</variable>
</variables>
<slots>
    <slot access="protected">closeEvent( QCloseEvent * e )</slot>
    <slot>fileNew()</slot>
    <slot>fileExit()</slot>
    <slot>fileMakeDefault()</slot>
    <slot>filePreferences()</slot>
    <slot>fileOpen()</slot>
    <slot returnType="bool">save()</slot>
    <slot returnType="bool">saveAs()</slot>
    <slot>pwordAdd()</slot>
    <slot>pwordDelete()</slot>
    <slot>pwordEdit()</slot>
    <slot>pwordFetch()</slot>
    <slot>pwordFetchUser()</slot>
    <slot>onPwordListDblClicked( QListViewItem * item )</slot>
    <slot>onPwordListRightClk( QListViewItem *, const QPoint &amp; point, int )</slot>
    <slot>helpAbout()</slot>
    <slot>helpManual()</slot>
    <slot>createNewSafe( const EncryptedString &amp; passphrase )</slot>
    <slot>fileOpenDefault()</slot>
    <slot>fileChangePassPhrase()</slot>
    <slot>lock()</slot>
    <slot>createGroup()</slot>
    <slot>setClearClipboardOnExit( bool yes )</slot>
    <slot>setLockOnMinimize( bool yes )</slot>
    <slot>showEvent( QShowEvent * )</slot>
</slots>
<functions>
    <function access="private" specifier="non virtual">init()</function>
    <function access="private" specifier="non virtual">destroy()</function>
    <function specifier="non virtual" returnType="bool">closeSafe()</function>
    <function returnType="bool">open( const char * filename, const EncryptedString &amp; passkey )</function>
    <function returnType="bool">open( const char * filename, const EncryptedString &amp; passkey, const char * type )</function>
    <function specifier="non virtual" returnType="Safe *">getSafe()</function>
    <function specifier="non virtual" returnType="int">getGeneratedPwordLength()</function>
    <function access="protected">savingEnabled( bool value )</function>
    <function returnType="const QString &amp;">getDefaultSafe()</function>
    <function returnType="bool">browseForSafe( QString &amp; filename, QString &amp; filter, bool saving )</function>
    <function returnType="bool">clearClipboardOnExit() const</function>
    <function returnType="bool">lockOnMinimize() const</function>
</functions>
<pixmapinproject/>
<layoutdefaults spacing="6" margin="11"/>
<includehints>
    <includehint>safelistview.h</includehint>
</includehints>
</UI>
